# Практическая работа №3: Обработка выборочных данных. Нахождение интервальных оценок параметров распределения. Проверка статистической гипотезы о нормальном распределении

Выполнили студенты гр. 2383 Борисов Иван Павлович и Иваницкий Илья Александрович. Вариант №26

## Цель работы

Получение практических навыков вычисления интервальных статистических оценок параметров распределения выборочных данных и проверки «справедливости» статистических гипотез.

## Основные теоретические положения

Интервал $(\theta^* - \delta;\ \theta^* + \delta)$ называется доверительным интервалом и является интервальной оценкой параметра $\theta$.

Доверительный интервал для математического ожидания при неизвестном СКО:

$\left( \bar{x}_в - \frac{t_{\gamma} S}{\sqrt{N}}, \bar{x}_в + \frac{t_{\gamma} S}{\sqrt{N}} \right)$, где $ t_\gamma $ — коэффициент Стьюдента.

Доверительный интервал для математического ожидания при известном СКО:

$(S(1 - q), S(1 + q))$, где $ q $ — табличный коэффициент, зависящий от объема выборки $ N $ и надежности $ \gamma $.

Статистической гипотезой называют гипотезу о виде неизвестного распледеления исследуемой случайной величиный или о значении параметров известных распределений.

Выдвинутую гипотезу называют нулевой или основной гипотезой и называют $H_0$.

Проверку статистической гипотезы проводят для принятого уровня значимости $\alpha$

При проверке статистических гипотез на основе выборочных данных вычисляются значения этих случайных величин (наблюденные значения), которые сопоставляются с некоторыми, так называемыми, критическими (зависящими от заданного уровня значимости $\alpha$) значениями этих случайных величин.

Критерий Пирсона:

$\chi^2_{набл} = \sum_{i=1}^{K} \frac{(n_i - n_i')^2}{n_i'}$

Здесь $n_i'$ — теоретические частоты — частоты, с которыми нормально распределенная величина с параметрами $\bar{x}_в$ и $S$ попадала бы в $i$-й интервал интервального ряда при проведении $N$ испытаний.

Формула для вычисления теоретических частот:

$n_i' = N \cdot p_i$, где $i = 1, 2, 3, ..., K$

Формула для вычисления вероятности попадания случайной величины в каждый $i$-й интервал интервального ряда:

$p_i = \Phi (z_{i+1}) - \Phi (z_{i})$, где $i = 1, 2, 3, ..., K$

Формула для вычисления значений границ интервалов интервального ряда:

$z_i = \frac{x_i - \bar{x}_в}{S}$, где $i = 2, 3, ..., K$

Функция Лапласа:

$\Phi (z) = \frac{1}{\sqrt{2\pi}} \int_0^z e^{-\frac{t^2}{2}} dt$

Число степеней свободы:

$k = K - 3$

Областью принятия нулевой гипотезы называют совокупность значения критерия, при которых нулевую гипотезу принимают.

Критической областью называют совокупность значения критерия, при которых нулевую гипотезу отвергают.

Гипотеза о нормальном распределении принимается, если выполняется:

$ \chi^2_{набл} \leq \chi^2_{крит} $

## Постановка задачи

*   На основе выборки (из работы №2) и при заданной доверительной вероятности (надежности) определить границы доверительных интервалов для параметров генеральной совокупности: математического ожидания $M(X)$ и среднего квадратического отклонения $\sigma(X)$.
*   Применить критерий $\chi^2$ (Пирсона) для проверки гипотезы о том, что исследуемая случайная величина подчиняется нормальному закону распределения.
*   Дать содержательную интерпретацию полученным числовым характеристикам.

## Выполнение работы

In [8]:
!pip install numpy scipy tabulate

import math as m
import numpy as np
from scipy import stats
from tabulate import tabulate


[notice] A new release of pip is available: 25.0.1 -> 26.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


Defaulting to user installation because normal site-packages is not writeable


## Задание 1
### Доверительный интервал для математического ожидания

In [9]:
# Данные
N = 102
x_v = 10.245882
s = 0.952782

gamma_values = [0.95, 0.99]

# коэффициенты Стьюдента (df = 101)
t_values = [
    stats.t.ppf((1+0.95)/2, 101),
    stats.t.ppf((1+0.99)/2, 101)
]

delta_values = [round(t * s / m.sqrt(N), 6) for t in t_values]
border_values = [(round(x_v - d, 6), round(x_v + d, 6)) for d in delta_values]

for i in range(len(gamma_values)):
    print(f"γ = {gamma_values[i]}")
    print(f"δ = {delta_values[i]}")
    print(f"Доверительный интервал: ({border_values[i][0]}, {border_values[i][1]})\n")

γ = 0.95
δ = 0.187144
Доверительный интервал: (10.058738, 10.433026)

γ = 0.99
δ = 0.247678
Доверительный интервал: (9.998204, 10.49356)



В данном блоке производится построение доверительных интервалов
для математического ожидания генеральной совокупности.

Используется распределение Стьюдента, поскольку истинное значение
среднеквадратического отклонения генеральной совокупности неизвестно
и заменяется выборочной оценкой.

Полученные интервалы показывают диапазон значений, в котором
с заданной доверительной вероятностью может находиться
математическое ожидание исследуемой случайной величины.

При увеличении доверительной вероятности γ возрастает значение
критического коэффициента распределения Стьюдента,
что приводит к увеличению предельной ошибки δ
и расширению доверительного интервала.

## Задание 2
### Доверительный интервал для среднеквадратичного отклонения

In [10]:
# коэффициенты q для N ≈ 100
q_values = [0.143, 0.198]

borders = [(round(s * (1 - q), 6), round(s * (1 + q), 6)) for q in q_values]

for i in range(len(gamma_values)):
    print(f"γ = {gamma_values[i]}")
    print(f"Доверительный интервал: ({borders[i][0]}, {borders[i][1]})\n")

γ = 0.95
Доверительный интервал: (0.816534, 1.08903)

γ = 0.99
Доверительный интервал: (0.764131, 1.141433)



Так как табличные данные не содержали значения для объема выборки 102, коэффициент $q$ принимался по ближайшему имеющемуся значению (для $N=100$).

В данном блоке вычисляются доверительные интервалы
для среднеквадратического отклонения генеральной совокупности.

Поскольку распределение дисперсии связано с распределением χ²,
для построения интервала используются табличные коэффициенты q,
зависящие от объема выборки и уровня доверия.

Полученные границы интервала показывают возможные значения
истинного среднеквадратического отклонения генеральной совокупности.

Как и в предыдущем случае, увеличение доверительной вероятности
приводит к расширению интервала, что связано
с увеличением допустимой погрешности оценки параметра.

## Задание 3
### Проверка гипотезы о нормальном распределении

In [36]:
# Функция Лапласа
def Phi(z):
    return stats.norm.cdf(z) - 0.5


intervals = [
    {"left": 8.4, "right": 8.95, "n_i": 1},
    {"left": 8.95, "right": 9.5, "n_i": 22},
    {"left": 9.5, "right": 10.05, "n_i": 32},
    {"left": 10.05, "right": 10.6, "n_i": 18},
    {"left": 10.6, "right": 11.15, "n_i": 10},
    {"left": 11.15, "right": 11.7, "n_i": 9},
    {"left": 11.7, "right": 12.25, "n_i": 4},
    {"left": 12.25, "right": 12.8, "n_i": 6}
]


rows = []
table_data = []

for i, inter in enumerate(intervals):

    left = inter["left"]
    right = inter["right"]
    n_i = inter["n_i"]

    if i == 0:
        z_l = -np.inf
        phi_l = -0.5
    else:
        z_l = (left - x_v) / s
        phi_l = Phi(z_l)

    if i == len(intervals) - 1:
        z_r = np.inf
        phi_r = 0.5
    else:
        z_r = (right - x_v) / s
        phi_r = Phi(z_r)

    p_i = round(phi_r - phi_l, 4)
    n_i_theor = round(N * p_i, 2)

    rows.append({
        "interval": f"[{left}, {right})",
        "n_i": n_i,
        "p_i": p_i,
        "n_i_theor": n_i_theor
    })

    table_data.append([
        i+1,
        left,
        right,
        n_i,
        "-∞" if z_l == -np.inf else f"{z_l:.3f}",
        "+∞" if z_r == np.inf else f"{z_r:.3f}",
        f"{phi_l:.4f}",
        f"{phi_r:.4f}",
        p_i,
        n_i_theor
    ])


headers = ["i", "x_i", "x_i+1", "n_i", "z_i", "z_i+1", "Φ(z_i)", "Φ(z_i+1)", "p_i", "n_i'"]

print(tabulate(table_data, headers=headers, tablefmt="grid"))

+-----+-------+---------+-------+--------+---------+----------+------------+--------+--------+
|   i |   x_i |   x_i+1 |   n_i | z_i    | z_i+1   |   Φ(z_i) |   Φ(z_i+1) |    p_i |   n_i' |
+=====+=======+=========+=======+========+=========+==========+============+========+========+
|   1 |  8.4  |    8.95 |     1 | -∞     | -1.360  |  -0.5    |    -0.4131 | 0.0869 |   8.86 |
+-----+-------+---------+-------+--------+---------+----------+------------+--------+--------+
|   2 |  8.95 |    9.5  |    22 | -1.360 | -0.783  |  -0.4131 |    -0.2831 | 0.13   |  13.26 |
+-----+-------+---------+-------+--------+---------+----------+------------+--------+--------+
|   3 |  9.5  |   10.05 |    32 | -0.783 | -0.206  |  -0.2831 |    -0.0814 | 0.2017 |  20.57 |
+-----+-------+---------+-------+--------+---------+----------+------------+--------+--------+
|   4 | 10.05 |   10.6  |    18 | -0.206 | 0.372   |  -0.0814 |     0.1449 | 0.2264 |  23.09 |
+-----+-------+---------+-------+--------+--------

Так как в 7 и 8 интервалах теоретическая частота меньше 5, следовательно, нужно объединить 7 и 8 интервалы.

In [37]:
merged = rows[:-2]  # все интервалы кроме последних двух
# объединяем последние два
last1 = rows[-2].copy()
last2 = rows[-1].copy()
last1["interval"] = last1["interval"] + " + " + last2["interval"]
last1["n_i"] += last2["n_i"]
last1["p_i"] += last2["p_i"]
last1["n_i_theor"] += last2["n_i_theor"]
merged.append(last1)

# === Вывод итоговой таблицы ===
table_data = []
for i, row in enumerate(merged):
    table_data.append([
        i+1,
        row["interval"],
        row["n_i"],
        round(row["p_i"], 4),
        round(row["n_i_theor"], 2)
    ])

headers = ["i", "Интервал", "n_i (факт)", "p_i (теор.)", "n_i' (теор.)"]
print(tabulate(table_data, headers=headers, tablefmt="grid"))

+-----+-------------------------------+--------------+---------------+----------------+
|   i | Интервал                      |   n_i (факт) |   p_i (теор.) |   n_i' (теор.) |
+=====+===============================+==============+===============+================+
|   1 | [8.4, 8.95)                   |            1 |        0.0869 |           8.86 |
+-----+-------------------------------+--------------+---------------+----------------+
|   2 | [8.95, 9.5)                   |           22 |        0.13   |          13.26 |
+-----+-------------------------------+--------------+---------------+----------------+
|   3 | [9.5, 10.05)                  |           32 |        0.2017 |          20.57 |
+-----+-------------------------------+--------------+---------------+----------------+
|   4 | [10.05, 10.6)                 |           18 |        0.2264 |          23.09 |
+-----+-------------------------------+--------------+---------------+----------------+
|   5 | [10.6, 11.15)           

После вычисления теоретических частот было проверено условие
применимости критерия Пирсона.

Если ожидаемая частота некоторого интервала меньше 5,
то такой интервал объединяется с соседним.

В результате объединения формируется новый набор интервалов,
для которых выполняется условие применимости критерия.

Полученные объединённые интервалы используются далее
для вычисления статистики критерия χ².

## Задание 4

Докажем, что

$\chi^2_{набл} = \sum_{i=1}^{k} \frac{n_i^2}{n'_i} - n$

Доказательство:

$\chi^2_{набл} = \sum_{i=1}^{k} \frac{(n_i - n'_i)^2}{n'_i} = \sum_{i=1}^{k} \frac{n_i^2 - 2n_i n'_i + (n'_i)^2}{n'_i} = \sum_{i=1}^{k} \frac{n_i^2}{n'_i} - 2\sum_{i=1}^{k} n_i + \sum_{i=1}^{k} n'_i = \sum_{i=1}^{k} \frac{n_i^2}{n'_i} - 2n + n = \sum_{i=1}^{k} \frac{n_i^2}{n'_i} - n$

In [40]:
chi_table = []

sum_chi = 0
sum_n_sq_div = 0

for i, row in enumerate(merged, 1):

    n_i = row["n_i"]
    p_i = row["p_i"]
    n_i_theor = row["n_i_theor"]

    diff_sq = round((n_i - n_i_theor) ** 2, 4)
    chi = round(diff_sq / n_i_theor, 4)
    n_sq_div = round((n_i ** 2) / n_i_theor, 4)

    sum_chi += chi
    sum_n_sq_div += n_sq_div

    chi_table.append([
        i,
        row["interval"],
        n_i,
        p_i,
        n_i_theor,
        diff_sq,
        chi,
        n_i ** 2,
        n_sq_div
    ])


chi_table.append([
    "",
    "Σ",
    sum(r[2] for r in chi_table),
    sum(r[3] for r in chi_table),
    sum(r[4] for r in chi_table),
    "-",
    sum_chi,
    "-",
    sum_n_sq_div
])

headers = [
    "i",
    "[x_i, x_i+1)",
    "n_i",
    "p_i",
    "n_i'",
    "(n_i-n_i')^2",
    "(n_i-n_i')^2/n_i'",
    "n_i^2",
    "n_i^2/n_i'"
]

print(tabulate(chi_table, headers=headers, tablefmt="grid"))

print("\nНаблюдаемое значение χ² =", round(sum_chi,4))

chi_obs = round(sum_n_sq_div - N, 4)

print("Наблюдаемое значение по формуле χ² =", chi_obs)

+-----+-------------------------------+-------+--------+--------+----------------+---------------------+---------+--------------+
|   i | [x_i, x_i+1)                  |   n_i |    p_i |   n_i' | (n_i-n_i')^2   |   (n_i-n_i')^2/n_i' | n_i^2   |   n_i^2/n_i' |
+=====+===============================+=======+========+========+================+=====================+=========+==============+
|   1 | [8.4, 8.95)                   |     1 | 0.0869 |   8.86 | 61.7796        |              6.9729 | 1       |       0.1129 |
+-----+-------------------------------+-------+--------+--------+----------------+---------------------+---------+--------------+
|   2 | [8.95, 9.5)                   |    22 | 0.13   |  13.26 | 76.3876        |              5.7608 | 484     |      36.5008 |
+-----+-------------------------------+-------+--------+--------+----------------+---------------------+---------+--------------+
|   3 | [9.5, 10.05)                  |    32 | 0.2017 |  20.57 | 130.6449       |        

Оба вычисленных наблюдаемых значения сошлись.

## Задание 5

In [41]:
K = len(merged)

df = K - 3
alpha = 0.05

chi_crit = round(stats.chi2.ppf(1 - alpha, df), 4)

print("Число степеней свободы =", df)
print("Критическое значение χ² =", chi_crit)

if sum_chi <= chi_crit:
    print("Нулевая гипотеза принимается")
else:
    print("Нулевая гипотеза отвергается")

Число степеней свободы = 4
Критическое значение χ² = 9.4877
Нулевая гипотеза отвергается


Так как наблюдаемое значение критерия $\chi^2_{\text{набл}}$ оказалось **больше критического** $\chi^2_{\text{кр}}$, нулевая гипотеза $H_0$ о нормальном распределении **отвергается** на уровне значимости $\alpha = 0{,}05$.

Это означает, что различия между эмпирическими и теоретическими частотами являются **статистически значимыми**. Следовательно, выборочные данные **не согласуются** с гипотезой о нормальном распределении исследуемой случайной величины.



## Выводы

В ходе выполнения практической работы на основе выборочных данных были построены интервальные оценки параметров генеральной совокупности.

Были найдены доверительные интервалы для математического ожидания и среднеквадратического отклонения генеральной совокупности при уровнях надежности $\gamma = 0{,}95$ и $\gamma = 0{,}99$:

| Параметр | $\gamma = 0{,}95$ | $\gamma = 0{,}99$ |
|:--------:|:------------------:|:------------------:|
| $M(X)$ | (10.059, 10.433) | (9.998, 10.494) |
| $\sigma(X)$ | (0.817, 1.089) | (0.764, 1.141) |

Установлено, что увеличение доверительной вероятности приводит к расширению доверительных интервалов, что свидетельствует о снижении точности оценки при повышении надежности.

Далее была проверена гипотеза о нормальном распределении исследуемого признака с помощью критерия Пирсона $\chi^2$. Перед применением критерия интервалы были объединены таким образом, чтобы теоретические частоты удовлетворяли условию применимости критерия ( $n_i' \geq 5$ ).

В результате расчетов получено наблюдаемое значение $\chi^2_{\text{набл}} = 26{,}56$. Критическое значение при уровне значимости $\alpha = 0{,}05$ и числе степеней свободы $k = 4$ составило $\chi^2_{\text{кр}} = 9{,}49$.

Так как $\chi^2_{\text{набл}} > \chi^2_{\text{кр}}$, нулевая гипотеза о нормальном распределении **отвергается**. Это означает, что различия между эмпирическими и теоретическими частотами являются статистически значимыми, и выборочные данные **не согласуются** с предположением о нормальном законе распределения исследуемой случайной величины.